### ISSUE TITLE
Selectors don't play nicely with OverloadedStrings

### ISSUE THREAD

**Issue by 5outh** [fimad/scalpel#42]
For example, this:

```hs
imageURL :: Scraper BL.ByteString BL.ByteString
imageURL = attr "src" $ "img"
```

spits out the following error (with `OverloadedStrings` turned on):

```
/Users/Ben/projects/amazon-experiments/src/Main.hs:224:12:
    No instance for (Selectable r0) arising from a use of ‘attr’
    The type variable ‘r0’ is ambiguous
    Note: there are several potential instances:
      instance Selectable Text.HTML.Scalpel.Any
        -- Defined in ‘scalpel-0.3.1:Text.HTML.Scalpel.Internal.Select.Types’
      instance Selectable Selector
        -- Defined in ‘scalpel-0.3.1:Text.HTML.Scalpel.Internal.Select.Types’
      instance Selectable String
        -- Defined in ‘scalpel-0.3.1:Text.HTML.Scalpel.Internal.Select.Types’
    In the expression: attr "src"
    In the expression: attr "src" $ "img"
    In an equation for ‘imageURL’: imageURL = attr "src" $ "img"
/Users/Ben/projects/amazon-experiments/src/Main.hs:224:25:
    No instance for (Data.String.IsString r0)
      arising from the literal ‘"img"’
    The type variable ‘r0’ is ambiguous
    Note: there are several potential instances:
      instance Data.String.IsString
                 aeson-0.11.2.1:Data.Aeson.Types.Internal.Value
        -- Defined in ‘aeson-0.11.2.1:Data.Aeson.Types.Internal’
      instance (a ~ ByteString) =>
               Data.String.IsString
                 (attoparsec-0.13.0.2:Data.Attoparsec.ByteString.Internal.Parser a)
        -- Defined in ‘attoparsec-0.13.0.2:Data.Attoparsec.ByteString.Char8’
      instance (a ~ text-1.2.2.1:Data.Text.Internal.Text) =>
               Data.String.IsString
                 (attoparsec-0.13.0.2:Data.Attoparsec.Text.Internal.Parser a)
        -- Defined in ‘attoparsec-0.13.0.2:Data.Attoparsec.Text.Internal’
      ...plus 14 others
    In the second argument of ‘($)’, namely ‘"img"’
    In the expression: attr "src" $ "img"
    In an equation for ‘imageURL’: imageURL = attr "src" $ "img"
```

The way around this is to provide an explicit type for `"img"` and `"src"`, e.g:

```hs
imageURL :: Scraper BL.ByteString BL.ByteString
imageURL = attr ("src" :: String) $ ("img" :: String)
```

which is a little tedious. I realize this is a limitation of the type system, it's not like the initial problem can be fixed outright. In practice, I've created a bunch of declarations like this:

```hs
src_ :: String
src_ = "src"

img_ :: String
img_ = "img"

-- and so on...
```

And used them like so:

```hs
imageURL :: Scraper BL.ByteString BL.ByteString
imageURL = attr src_ $ img_
```

It might be nice to provide some selectors for common HTML elements and attributes in the library itself for this purpose. What do you think?

**Comment by fimad** [comment]
The problem of using scalpel with overloaded strings should be solved in the (eventually) upcoming 0.4.0 release. Playing nicely with overloaded strings required breaking some of the existing API (see #29) so I've been waiting till all planned breaking API changes have landed before pushing the update. But since its been awhile since that change has landed I'm thinking I might as well just push the release now...

**Comment by fimad** [comment]
Closing this out as a duplicate. Please feel free to reopen if the approach taken in #40 does not cover your use case.
